src/graph/connect.cc

```diff
@@ -627,6 +627,10 @@ ncclResult_t ncclTopoPostset(struct ncclComm* comm, int* firstRanks, int* treePa
   // Only use full MAXCHANNELS for gfx94x
   int maxChannels = IsArchMatch(comm->topo->nodes[GPU].nodes[0].gpu.gcn, "gfx94") ? MAXCHANNELS : 2*CHANNEL_LIMIT;
 
+  if (graphs[NCCL_ALGO_RING]->nIntraChannels > 0 || comm->nNodes > 1) {
```

Comment:
why nNodes>1? single node is not applicable here?

Reply:
I set `maxChannel=64` for `nNodes>1` or `nIntraChannels >1` (can be true for single node also).

Reply:
got it